### ISSUE TITLE
Support parsing of mixed precision kernels with new metadata approach

### ISSUE THREAD

**Issue by rupertford** [stfc/PSyclone#2236]
The new PSyIR metadata approach does not support LFRic mixed precision kernels and fails with the following error:

```
Expected kernel metadata to be stored in the PSyIR as an UnknownFortranType, but found StructureType.
```

This is due to the fact that the `contains` part of the metadata is missing in mixed precision kernels.

**Comment by rupertford** [comment]
Created branch [2236_mixed_precision_metadata](https://github.com/stfc/PSyclone/tree/2236_mixed_precision_metadata)

**Comment by rupertford** [comment]
OK, so I forgot that there was a reason we could not do this - the compiler does not like it :-(

**Comment by rupertford** [comment]
Oh dear. When we parse the metadata into PSyIR and end up with a StructureType the input and output code is ...
```
  type, public, extends(kernel_type) :: matrix_vector_kernel_type
    private
    type(arg_type) :: meta_args(3) = (/                                    &
         arg_type(GH_FIELD,    GH_REAL, GH_INC,  ANY_SPACE_1),             &
         arg_type(GH_FIELD,    GH_REAL, GH_READ, ANY_SPACE_2),             &
         arg_type(GH_OPERATOR, GH_REAL, GH_READ, ANY_SPACE_1, ANY_SPACE_2) &
         /)
    integer :: operates_on = CELL_COLUMN
  end type
```
```
  type, public :: matrix_vector_kernel_type
    type(arg_type), dimension(3), private :: meta_args
    integer, private :: operates_on
  end type matrix_vector_kernel_type
```

**Comment by rupertford** [comment]
The problem of losing data values has been fixed in #2255. A new issue #2347 has been added to capture the metadata in LFRic PSyIR so I am closing this issue.
